src/gpu/intel/jit/ir/core.hpp

```diff
@@ -529,9 +600,28 @@ class type_t {
         return copy;
     }
 
-    type_t with_ptr() const {
+    type_t with_ptr(bool is_slm = false) const {
```

Comment:
`is_slm` flag here is not aligned with `packed()` and `simd()`. I think we can similarly add `type_t::slm()` to achieve the same and align with other API.